presto-main/src/main/java/com/facebook/presto/server/BasicQueryInfo.java

```diff
@@ -89,6 +93,7 @@ public BasicQueryInfo(QueryInfo queryInfo)
         this(queryInfo.getQueryId(),
                 queryInfo.getSession(),
                 queryInfo.getState(),
+                queryInfo.getErrorCode() != null ? StandardErrorCode.toErrorType( queryInfo.getErrorCode().getCode()) : null,
```

Comment:
Move this to `QueryInfo` and add a `private static` helper `getErrorType(@Nullable ErrorCode code)`:

```java
getErrorType(queryInfo.getErrorCode())